### ISSUE TITLE
Create a single way of setting user tag values in W3S

### ISSUE THREAD

**Issue by adamalton** [web3-storage/web3.storage#1195]
We should have one - and only one - way of setting values in the `user_tag` table. This is for a couple of reasons:

1. The `user_tags` table is deliberately structured so that we keep the history of the tags; when we want to add a new tag value we should set the `deleted_at` column in the existing row (if any) to `now` and then add a new row, not update the existing row. So we should have a function for setting tag values which encapsulates this logic and ensures that we always set values this way.
2. There are several cases in which we want to email the user when we change the value of a particular tag, specifically:
    * When we set their account to restricted.
    * When we un-restrict their account (if we ever do this).
    * When we increase their storage quota.

So it seems to make sense that there should be a single method through which we set user tags, probably in the `db` package.

**In scope**. 
* Adding a function to do this.

**Out of scope**. 
* Sending those emails.

## Notes
Current implementation of `createUserTag` is suboptimal and could also lead to an incongruent state. Can we please make sure we improve it as part of this ticket?

- It's not transactional, we might end up having deleted a tag without adding the new one.
- The function is currently doing 2 api calls, not great for performance.
- This is currently not tested

**Comment by trigramdev9** [comment]
> The user_tags table is deliberately structured so that we keep the history of the tags; when we want to add a new tag value we should set the deleted_at column in the existing row (if any) to now and then add a new row, not update the existing row. So we should have a function for setting tag values which encapsulates this logic and ensures that we always set values this way.

This is already occurring.

**Comment by JeffLowe** [comment]
Just to clarify surrounding notifications:
We did previously discuss triggering user email notifications from within admin.storage with Leslie (Potato) in an early planning session.  However, there is no desire (thus no spec) for automated email notifications to users upon account blocking/unblocking.  

Given the fact that web3 storage increase requests from users will be sent via email to support (currently in mailto links, and eventually as a side effect of webform submission) which enables desired ability for possible back and forth communications between support and user, seems reasonable to not auto email the user upon limit increase.

**Comment by mbommerez** [comment]
### Actions

I will move this to the backlog. Though we understand there is no immediate desire to formalise user tags and their associated notifications in a single service, we do believe longer term this is needed. 

All in all, let's continue with the current approach as implemented in admin.storage by @trigramdev9  and push out an MVP and then we can consider the rest in a future iteration.

### POT feedback

>  it would be easier to kick off these type of emails (setting tags, removing tags etc.) directly from the admin.storage app.

* We think eventually automated transactional admin email notifications will be needed for this purpose as web3.storage scales. Ideally we were thinking of implementing the building blocks for this to be possible in the future.
* Also, there is the issue of duplicating email services for the same userbase. We already have an email service built in W3s for notifying users when they reach certain storage milestones. If we also trigger emails from admin.storage directly without a middle layer, it means you need to build a second email service in admin.storage, which will then not trigger logging (of the emails) on W3s side and which means that two email versions need to be maintained and kept in sync (so users receive the same type of emails across the board).

> This is already occurring.

Yes, we noticed this. The remark was more about formalising this logic and therefore preventing in the future for this to happen without that logic someplace else in admin.storage.

> However, there is no desire for automated email notifications to users upon account blocking/unblocking.

We believe in the future there might need to be as web3.storage scales and the constitution of the userbase changes.

> Given the fact that web3 storage increase requests from users will be sent via email to support (currently in mailto links, and eventually as a side effect of webform submission) which enables desired ability for possible back and forth communications between support and user, seems reasonable to not auto email the user upon limit increase.

Transactional emails are a good addition and a service for users, especially as we are looking at taking away the guesswork from users and make their storage with web3.storage more straightforward and predictable. Such an email can just confirm that the change has taken place, rather than just a manual message from an admin.

Hope this clarifies our thinking a bit! Happy to discuss further in a backlog refinement session.
